devices/Si7021/Si7021.cs

```diff
@@ -118,21 +161,19 @@ private RelativeHumidity GetHumidity()
         }
 
         /// <summary>
-        /// Get Si7021 Firmware Revision
+        /// Get Si7021 firmware revision.
         /// </summary>
-        /// <returns>Firmware Revision</returns>
-        private byte GetRevision()
+        /// <returns>The FirmwareRevision.</returns>
+        private FirmwareRevision GetRevision()
         {
             SpanByte writeBuff = new byte[2]
             {
-                (byte)Register.SI_REVISION_MSB, (byte)Register.SI_REVISION_LSB
+                (byte)Command.SI_REVISION_MSB, (byte)Command.SI_REVISION_LSB
             };
 
             _i2cDevice.Write(writeBuff);
-            // wait SCL free
-            Thread.Sleep(20);
 
-            return _i2cDevice.ReadByte();
+            return (FirmwareRevision)_i2cDevice.ReadByte();
```

Comment:
`return switch(_i2cDevice.ReadByte()) =>` with a new Version(1, 0), new Version(2, 0), etch should just make the job. You can keep the enum but private and remove fully the extension